website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -306,6 +310,16 @@ spec:
     - id: "subnet-0471ca205b8a129ae"
 ```
 
+## spec.associatePublicIpAddress
+
+a boolean field to control whether the instances created by karpenter for this node class will have an associated public IP address. 
+when not specified, the node will have public IP adders if it is launched in a subnet with public IP auto-assign enabled `MapPublicIpOnLaunch=true`.
+
+{{% alert title="Note" color="warning" %}}
+setting this field to true can cause instance launches to fail if the instance was configured with multiple EFAs. the two options can not be used together
+{{% /alert %}}
```

Comment:
```suggestion
A boolean field that controls whether instances created by Karpenter for this EC2NodeClass will have an associated public IP address. This overrides the `MapPublicIpOnLaunch` setting applied to the subnet the node is launched in. If this field is not set, the `MapPublicIpOnLaunch` field will be respected.

{{% alert title="Note" color="warning" %}}
If `associatePublicIPAddress` is set to `true` and a NodePool referencing that EC2NodeClass is used to provision a Node for a NodeClaim requesting multiple `vpc.amazonaws.com/efa` resources, the instance launch will fail. A public IP address may only be associated with a node at launch if a single network interface is configured. 
{{% /alert %}}
```

Could you also move this block to the end of this section so the order matches the EC2NodeClass example at the top of the page?